pkgs/applications/emulators/ruffle/unify-dasp-version.patch

```diff
@@ -0,0 +1,113 @@
+  "libc",
+@@ -1076,7 +1076,7 @@ dependencies = [
+  "dasp_peak",
+  "dasp_ring_buffer",
+  "dasp_rms",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+  "dasp_signal",
+  "dasp_slice",
+  "dasp_window",
+@@ -1091,7 +1091,7 @@ dependencies = [
+  "dasp_peak",
+  "dasp_ring_buffer",
+  "dasp_rms",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+ [[package]]
+@@ -1099,7 +1099,7 @@ name = "dasp_frame"
+ version = "0.11.0"
+ source = "git+https://github.com/RustAudio/dasp?rev=f05a703#f05a703d247bb504d7e812b51e95f3765d9c5e94"
+ dependencies = [
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+ [[package]]
+@@ -1109,7 +1109,7 @@ source = "git+https://github.com/RustAudio/dasp?rev=f05a703#f05a703d247bb504d7e8
+ dependencies = [
+  "dasp_frame",
+  "dasp_ring_buffer",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+ [[package]]
+@@ -1118,7 +1118,7 @@ version = "0.11.0"
+ source = "git+https://github.com/RustAudio/dasp?rev=f05a703#f05a703d247bb504d7e812b51e95f3765d9c5e94"
+ dependencies = [
+  "dasp_frame",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+ [[package]]
+@@ -1133,15 +1133,9 @@ source = "git+https://github.com/RustAudio/dasp?rev=f05a703#f05a703d247bb504d7e8
+ dependencies = [
+  "dasp_frame",
+  "dasp_ring_buffer",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+-[[package]]
+-name = "dasp_sample"
+-version = "0.11.0"
+-source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "0c87e182de0887fd5361989c677c4e8f5000cd9491d6d563161a8f3a5519fc7f"
+-
+ [[package]]
+ name = "dasp_sample"
+ version = "0.11.0"
+@@ -1158,7 +1152,7 @@ dependencies = [
+  "dasp_peak",
+  "dasp_ring_buffer",
+  "dasp_rms",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+  "dasp_window",
+ ]
+ 
+@@ -1168,7 +1162,7 @@ version = "0.11.0"
+ source = "git+https://github.com/RustAudio/dasp?rev=f05a703#f05a703d247bb504d7e812b51e95f3765d9c5e94"
+ dependencies = [
+  "dasp_frame",
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+ [[package]]
+@@ -1176,7 +1170,7 @@ name = "dasp_window"
+ version = "0.11.0"
+ source = "git+https://github.com/RustAudio/dasp?rev=f05a703#f05a703d247bb504d7e812b51e95f3765d9c5e94"
+ dependencies = [
+- "dasp_sample 0.11.0 (git+https://github.com/RustAudio/dasp?rev=f05a703)",
++ "dasp_sample",
+ ]
+ 
+ [[package]]
+diff --git a/Cargo.toml b/Cargo.toml
+index c3d25e662..b4110a420 100644
+--- a/Cargo.toml
++++ b/Cargo.toml
+@@ -71,3 +71,6 @@ inherits = "release"
+ 
+ [profile.web-wasm-extensions]
+ inherits = "release"
++
++[patch.crates-io]
++dasp_sample = { git = "https://github.com/RustAudio/dasp", rev = "f05a703" }
```

Comment:
```suggestion
+dasp_sample = { git = "https://github.com/RustAudio/dasp", rev = "f05a703d247bb504d7e812b51e95f3765d9c5e94" }
```

Reply:
Tried this, but cargo treats this as distinct from the existing reference to dasp_sample at `f05a703`, so it still fails to build. I could make the existing git reference use a long hash if you think that would be better, in exchange for a slightly larger diff. WDYT?

Reply:
We should use the full sha because the short sha could become ambiguous in the future which will make github return a 404.

Reply:
OK. I moved both instances of dasp to use the full SHA sum, including the upstream reference which already used a truncated hash.